src/org/parosproxy/paros/model/SiteMap.java

```diff
@@ -745,5 +743,26 @@ private void clearParentFilter (SiteNode parent) {
 			clearParentFilter(parent.getParent());
 		}
 	}
+	
+	@Override
+	public void removeNodeFromParent(MutableTreeNode node) {
+		super.removeNodeFromParent(node);
+		if(node.getParent() == getRoot()) {
+			ZAP.getEventBus().publishSyncEvent(SiteMapEventPublisher.getPublisher(), 
+	        		new Event(SiteMapEventPublisher.getPublisher(), SiteMapEventPublisher.SITE_REMOVED_EVENT, new Target((SiteNode)node)));
+		} else {
+			ZAP.getEventBus().publishSyncEvent(SiteMapEventPublisher.getPublisher(), 
+	        		new Event(SiteMapEventPublisher.getPublisher(), SiteMapEventPublisher.SITE_NODE_REMOVED_EVENT, new Target((SiteNode)node)));

```

Comment:
This event should be published in both cases, if it is a site or not, otherwise listeners that check only the SITE_NODE events (like the Attack mode) would not know that the node (that represents the site) was added.